blockchain/validate.go

```diff
@@ -1314,32 +1275,44 @@ func (b *BlockChain) checkBlockHeaderContext(header *wire.BlockHeader, prevNode
 	return nil
 }
 
-// checkCoinbaseUniqueHeightWithAddress checks to ensure that for all blocks
-// height > 1 the coinbase contains the height encoding to make coinbase hash
-// collisions impossible. This is the old function that assumes a treasury
-// address.
-func checkCoinbaseUniqueHeightWithAddress(blockHeight int64, minReqOutputs, index int, block *dcrutil.Block) error {
-	// When treasury is disabled, Coinbase output 0 is the project subsidy,
-	// and output 1 is height + extranonce, so at least two outputs must
-	// exist. When the treasury is enabled the coinbase treasury payout has
-	// been moved to the stake tree and therefore we only need one output
-	// which encodes the height + extranonce in output 0.
+// checkCoinbaseUniqueHeight checks to ensure that for all blocks height > 1 the
+// coinbase contains the height encoding to make coinbase hash collisions
+// impossible.
+func checkCoinbaseUniqueHeight(blockHeight int64, block *dcrutil.Block, treasuryEnabled bool) error {
+	// Block 0 and 1 are special and don't need the coinbase height checks.
+	if blockHeight < 2 {
+		return nil
+	}
+
+	// Choose the correct output that encodes the height depending on whether
+	// or not the treasury agenda is active.
+	//
+	// Prior to activation of the agenda, output 0 is the project subsidy and
+	// output 1 encodes the height.  Once the agenda is active, the project
+	// subsidy is moved to the treasurybase in the stake tree and thus output 0
+	// then encodes the height.
+	nullDataOutIdx := 1
+	if treasuryEnabled {
+		nullDataOutIdx = 0
+	}
 
+	// There must be at least enough outputs to contains the one that encodes
```

Comment:
s/ ..outputs to contains / ..outputs to contain

Reply:
Nice catch.  Updated.